target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-rax120.dts

```diff
@@ -0,0 +1,1010 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/dts-v1/;
+/* Copyright (c) 2017, The Linux Foundation. All rights reserved.
+ */
+#include "ipq8074.dtsi"
+#include "ipq8074-ess.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
+#include "ipq8074-512m.dtsi"
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	#address-cells = <0x2>;
+	#size-cells = <0x2>;
+	model = "Netgear RAX120";
+	compatible = "netgear,rax120", "qcom,ipq8074";
+	interrupt-parent = <&intc>;
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		serial1 = &blsp1_uart3;
+		label-mac-device = &dp1;
+		ethernet0 = &dp1;
+		ethernet1 = &dp2;
+		ethernet2 = &dp3;
+		ethernet3 = &dp4;
+		ethernet4 = &dp5;
+		ethernet5 = &dp6;
+	};
+
+	chosen {
+		stdout-path = "serial0";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			gpios = <&tlmm 57 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		rfkill {
+			label = "rfkill";
+			gpios = <&tlmm 25 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+	};
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		sck-gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+		mosi-gpios = <&tlmm 19 GPIO_ACTIVE_HIGH>;
+
+
+		led_gpio: led_gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <2>;
+			spi-max-frequency = <1000000>;
+			enable-gpios = <&tlmm 20 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led0 {
+			label = "Power";
+			gpios = <&led_gpio 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_wifi_son_orange";
+		};
+
+		led1 {
+			label = "24g_on";
+			gpios = <&led_gpio 1 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_wifi_son_orange";
+		};
+
+		led2 {
+			label = "5g_on";
+			gpios = <&led_gpio 2 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_wifi_son_blue";
+		};
+
+		led3 {
+			label = "USB2";
+			gpios = <&led_gpio 3 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		led4 {
+			label = "USB1";
+			gpios = <&led_gpio 4 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_5gl";
+		};
+
+		led5 {
+			label = "Internet";
+			gpios = <&led_gpio 5 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_5gh";
+		};
+
+		led6 {
+			label = "MultiGig:Green";
+			gpios = <&led_gpio 6 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_lan";
+		};	
+
+		led7 {
+			label = "MultiGig:Red";
+			gpios = <&led_gpio 10 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_lan";
+		};
+
+		led8 {
+			label = "MultiGig:White";
+			gpios = <&led_gpio 11 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_lan";
+		};
+
+		led9 {
+			label = "WPS";
+			gpios = <&tlmm 40 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "led_usb2_enable";
+		};
+	};
+
+	partitions {
```

Comment:
This is completely wrong, what are partitions doing here?
They must be under the device that actually carries them

Reply:
That was a mistake on my part. I was reviewing the dt-binding documentation for the correct placement and kept running into issues. I'll review again and make corrections.

Reply:
Well, you already have one partition table under the NAND which is the correct place to put it.

Reply:
Under NAND is how I intended although please confirm if this is correct under the context below. I was just trying different locations in DT after reading this. 

From: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/mtd/partition.txt?h=v5.15

Deprecated: partitions defined in flash node
============================================

For backwards compatibility partitions as direct subnodes of the flash device are
supported. This use is discouraged.
NOTE: also for backwards compatibility, direct subnodes that have a compatible
string are not considered partitions, as they may be used for other bindings.

Reply:
It's written a bit confusing, what is deprecated is putting the partition nodes directly under the storage node.
So like this:
```
nand@0 {
	partition@0 {
		label = "test";
		reg = <0x0 0x10000>;
	};
};
```

You must have a partitions subnode under storage and then partitions are subnode of it.

Reply:
Ok cool. Yeah they could have done a better job on the language. I'll get it cleaned up.